server/utils/cloudFunctions/functions/test/algoliaTests/algoliaIntegrationTests.test.js

```diff
@@ -0,0 +1,218 @@
+              chai.expect(content.data)
+                  .to.deep.equal({advertiser: "e", startDate: "2020-10-15"});
+            }).catch(err => console.log(err));
+            
+            done(); 
+          }, TIMEOUT_10S);
+        });
+      });
+    }).timeout(TIMEOUT_15S);
+
+    it("should propogate a firestore document update to " + 
+       "the proper algolia record", function(done) {
+      const adOne = {advertiser: "f", startDate: "2019-10-15"};
+      const adTwo = {advertiser: "g", startDate: "2019-10-15"};
+
+      DEV_ADS_COLLECTION.doc("f").set(adOne).then(() => {
+        DEV_ADS_COLLECTION.doc("g").set(adTwo).then(() => {
+          DEV_ADS_COLLECTION.doc("f").update({startDate: "2020-10-15"}).then(() => {
+            setTimeout(function(){
+
+              // Verify that `adOne` was updated, `adTwo` was not. 
+              DEV_ADS_INDEX.getObject("f").then(content => {
+                chai.expect(content.data)
+                    .to.deep.equal({advertiser: "f", startDate: "2020-10-15"});
+              }).catch(err => console.log(err));
+              
+              DEV_ADS_INDEX.getObject("g").then(content => {
+                chai.expect(content.data).to.deep.equal(adTwo);                
+              }).catch(err => console.log(err));
+
+              done(); 
+            }, TIMEOUT_10S);
+          });
+        });
+      });
+    }).timeout(TIMEOUT_15S);
+  });
+
+  describe("test_deleteRecord", () => {
+    it("should delete an algolia record when a firestore " + 
+       "document is deleted", function(done) {
+      const ad = {advertiser: "h", startDate: "2019-10-15"};
+
+      DEV_ADS_COLLECTION.doc("h").set(ad).then(() => {
+        DEV_ADS_COLLECTION.doc("h").delete().then(() => {
+          setTimeout(function(){
+            DEV_ADS_INDEX.getObjects(["h"]).then(content => {
+
+              // Checking `ad` is deleted, so verify an error message
+              // exists, has the right message, and that index 0
+              // is null. Check existence because if doesn't, `.trim()`
+              // will yield an unhelpful error and obfuscate the true error.
+              chai.expect(content.message).to.exist;
+              chai.expect(content.message.trim())
+                  .to.be.equal("ObjectID h does not exist.");
+              chai.expect(content.results[0]).to.be.null;
+            }).catch(err => console.log(err));
+
+            done();
+          }, TIMEOUT_10S);
+        });
+      });
+    }).timeout(TIMEOUT_15S); 
+
+    it("should delete the right algolia record when a firestore document is deleted", function(done) {
+      const adOne = {advertiser: "i", startDate: "2019-10-15"};
+      const adTwo = {advertiser: "j", startDate: "2019-10-15"};
+
+      DEV_ADS_COLLECTION.doc("i").set(adOne).then(() => {
+        DEV_ADS_COLLECTION.doc("j").set(adTwo).then(() => {
+          DEV_ADS_COLLECTION.doc("i").delete().then(() => {
```

Comment:
if you're using alphabetical letters for testing, what happen if you ran out of letters? :)

Reply:
`.doc("z"), .doc("aa"), .doc("ab"),...`. I see your point, but I think alphabetical enumeration is the best worst option for easy indexing.